server/ERNI.PBA.Server.Host/Controllers/InvoiceImageController.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using ERNI.PBA.Server.DataAccess;
+using ERNI.PBA.Server.DataAccess.Model;
+using ERNI.PBA.Server.DataAccess.Repository;
+using ERNI.PBA.Server.Host.Model.InvoiceImage;
+using ERNI.PBA.Server.Host.Model.PendingRequests;
+using ERNI.PBA.Server.Host.Utils;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.StaticFiles;
+
+namespace ERNI.PBA.Server.Host.Controllers
+{
+    [Route("api/[controller]")]
+    [Authorize]
+    public class InvoiceImageController : Controller
+    {
+        private readonly IInvoiceImageRepository _invoiceImageRepository;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly PermissionHelper _permissionHelper;
+
+        public InvoiceImageController(IInvoiceImageRepository invoiceImageRepository,
+            IUnitOfWork unitOfWork,
+            PermissionHelper permissionHelper)
+        {
+            _invoiceImageRepository = invoiceImageRepository;
+            _unitOfWork = unitOfWork;
+            _permissionHelper = permissionHelper;
+        }
+
+        [HttpGet("images/{requestId}")]
+        [Authorize]
+        public async Task<IActionResult> GetInvoiceImages(int requestId, CancellationToken cancellationToken)
+        {
+            var userId = HttpContext.User.GetId();
+            if (!await _permissionHelper.IsUsersRequestOrAdmin(userId, requestId, cancellationToken))
+            {
+                return Unauthorized();
+            }
+            var imagesName = await _invoiceImageRepository.GetInvoiceImages(requestId, cancellationToken);
+            var result = imagesName.Select(image => new
+            {
+                Id = image.Id,
+                Name = image.Name
+            });
+            return Ok(result);
+        }
+
+        [HttpGet("image/{imageId}")]
+        [Authorize]
+        public async Task<IActionResult> GetInvoiceImageFile(int imageId,
+            CancellationToken cancellationToken)
+        {
+            var image = await _invoiceImageRepository.GetInvoiceImage(imageId, cancellationToken);
+            var userId = HttpContext.User.GetId();
+            if (!await _permissionHelper.IsUsersRequestOrAdmin(userId, image.RequestId, cancellationToken))
```

Comment:
You do not need a helper class there. Check deleteRequest method:
var user = HttpContext.User;
            if (!user.IsInRole(Roles.Admin) && user.GetId() != request.UserId)
            {
                return BadRequest("Access denied");
            }